src/createXcmTypes/util/foreignAssetMultiLocationIsInCacheOrRegistry.ts

```diff
@@ -2,30 +2,54 @@
 
 import { ApiPromise } from '@polkadot/api';
 
+import { ASSET_HUB_CHAIN_ID } from '../../consts';
 import { BaseError, BaseErrorsEnum } from '../../errors';
 import { Registry } from '../../registry';
 import { ForeignAssetsInfo } from '../../registry/types';
 
-export const foreignAssetMultiLocationIsInRegistry = (
+export const foreignAssetMultiLocationIsInCacheOrRegistry = (
 	api: ApiPromise,
 	multilocationStr: string,
 	registry: Registry
 ): boolean => {
-	try {
-		const assetHubChainId = 1000;
+	if (!registry.cache[registry.relayChain][ASSET_HUB_CHAIN_ID]) {
```

Comment:
This should be added to the registry layer as part of the initialization call in the constructor.